values.go

```diff
@@ -38,6 +39,12 @@ func GetFlags(dir string) (map[string]string, error) {
 
 	if version, err := versionFromFile(dir); err != nil {
 		return nil, fmt.Errorf("failed to get version: %v", err)
+	} else if version == "" {
+		// ex: v0.1.2 0.1.2 0.1.2.dev3
+		matched, err := regexp.MatchString(`^v?(\d+\.){1,2}(\d+)`, gitSummary)
```

Comment:
@codeskyblue even in my team we use a.b.c.d style versioning. also I can't see how your regex matches something like -beta1 without cutting it away, which is necessary. I think a logic like this will heavily impair the transparency and the "does what it says on the label" aspect of the project. Therefore I'm against it.